src/test/java/seedu/address/logic/parser/AddAppointmentCommandParserTest.java

```diff
@@ -0,0 +1,201 @@
+
+        // whitespace only preamble
+        assertParseSuccess(parser, PREAMBLE_WHITESPACE
+                + PATIENT_NRIC_DESC
+                + DOCTOR_NRIC_DESC
+                + APPOINTMENT_START_TIME_DESC
+                + APPOINTMENT_END_TIME_DESC , new AddAppointmentCommand(expectedAppointment));
+
+    }
+
+    @Test
+    public void parse_repeatedValue_failure() {
+        String validExpectedAppointmentString = PATIENT_NRIC_DESC
+                + DOCTOR_NRIC_DESC
+                + APPOINTMENT_START_TIME_DESC
+                + APPOINTMENT_END_TIME_DESC;
+
+        // multiple patient nric
+        assertParseFailure(parser, PATIENT_NRIC_DESC + validExpectedAppointmentString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_PATIENT_NRIC));
+
+        // multiple doctor nric
+        assertParseFailure(parser, DOCTOR_NRIC_DESC + validExpectedAppointmentString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_DOCTOR_NRIC));
+
+        // multiple apointment start dates
+        assertParseFailure(parser, APPOINTMENT_START_TIME_DESC + validExpectedAppointmentString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_APPOINTMENT_START_TIME));
+
+        // multiple appointment end dates
+        assertParseFailure(parser, APPOINTMENT_END_TIME_DESC + validExpectedAppointmentString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_APPOINTMENT_END_TIME));
+
+        // multiple fields repeated
+        assertParseFailure(parser,
+                validExpectedAppointmentString + PATIENT_NRIC_DESC
+                        + DOCTOR_NRIC_DESC
+                        + APPOINTMENT_START_TIME_DESC
+                        + APPOINTMENT_END_TIME_DESC
+                        + validExpectedAppointmentString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_PATIENT_NRIC, PREFIX_DOCTOR_NRIC,
+                        PREFIX_APPOINTMENT_START_TIME, PREFIX_APPOINTMENT_END_TIME));
+
+        // invalid patient Nric
+        assertParseFailure(parser, INVALID_PATIENT_NRIC_DESC + validExpectedAppointmentString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_PATIENT_NRIC));
+
+        // invalid email
+        assertParseFailure(parser, INVALID_DOCTOR_NRIC_DESC + validExpectedAppointmentString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_DOCTOR_NRIC));
+
+        // invalid phone
+        assertParseFailure(parser, INVALID_APPOINTMENT_START_TIME_DESC + validExpectedAppointmentString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_APPOINTMENT_START_TIME));
+
+        // invalid address
+        assertParseFailure(parser, INVALID_APPOINTMENT_END_TIME_DESC + validExpectedAppointmentString,
+                Messages.getErrorMessageForDuplicatePrefixes(PREFIX_APPOINTMENT_END_TIME));
+
+        // valid value followed by invalid value
```

Comment:
missing testcase here?

Reply:
Its meant to describe the next set of test cases